### ISSUE TITLE
Update package webhook

### ISSUE THREAD

**Issue by WebFreak001** [dlang/dub-registry#412]
there should be generic and github compatible webhooks for triggering a package update (so users can configure them to trigger on tag/branch creation)

A secret would be stored with every package to prevent abuse.

When #410 is merged it shouldn't be a problem to add instructions for it to the package management UI

**Comment by wilzbach** [comment]
While this is a lot easier to implement, it's still quite some effort for the users (and they can mess it up by wrongly setting up the webhook), so probably the better way would be to do it either automatically via a GitHub Auth button or as GitHub integration.

**Comment by WebFreak001** [comment]
it would not be GitHub specific, it would work with GitLab and BitBucket and if we ever support custom git repositories, also with those

**Comment by wilzbach** [comment]
Hmm, maybe we can start with the hooks as you proposed them (as this should be fairly easy & straight-forward) and a future GitHub integration could set these hooks automatically?

**Comment by WebFreak001** [comment]
@wilzbach I fixed this issue (on another branch depending on #410 first being merged)

It also shows error messages if you misconfigured the hook on github:
![misconfigured github hook error](https://wfr.moe/fhHqgf.png)

The commit is above this comment

**Comment by WebFreak001** [comment]
duplicate of #275 and #10
